crates/ruma-common/src/events/room/message/relation_serde.rs

```diff
@@ -66,133 +67,157 @@ where
 {
     fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
     where
-        S: Serializer,
+        S: serde::Serializer,
     {
-        let json_repr = match self {
-            Relation::Reply { in_reply_to } => {
-                EventWithRelatesToJsonRepr::<C>::new(RelatesToJsonRepr {
-                    in_reply_to: Some(in_reply_to.clone()),
-                    ..Default::default()
-                })
-            }
-            Relation::Replacement(Replacement { event_id, new_content }) => {
-                EventWithRelatesToJsonRepr {
-                    relates_to: RelatesToJsonRepr {
-                        relation: Some(RelationJsonRepr::Replacement(ReplacementJsonRepr {
-                            event_id: event_id.clone(),
-                        })),
-                        ..Default::default()
-                    },
-                    new_content: Some(new_content.clone()),
-                }
-            }
-            Relation::Thread(Thread { event_id, in_reply_to, is_falling_back }) => {
-                EventWithRelatesToJsonRepr::new(RelatesToJsonRepr {
-                    in_reply_to: in_reply_to.clone(),
-                    relation: Some(RelationJsonRepr::ThreadStable(ThreadStableJsonRepr {
-                        event_id: event_id.clone(),
-                        is_falling_back: *is_falling_back,
-                    })),
-                })
-            }
-            Relation::_Custom => EventWithRelatesToJsonRepr::<C>::default(),
-        };
+        let (relates_to, new_content) = self.clone().into_parts();
 
-        json_repr.serialize(serializer)
+        EventWithRelatesToSerHelper { relates_to, new_content }.serialize(serializer)
     }
 }
 
-#[derive(Deserialize, Serialize)]
-struct EventWithRelatesToJsonRepr<C> {
-    #[serde(rename = "m.relates_to", default, skip_serializing_if = "RelatesToJsonRepr::is_empty")]
-    relates_to: RelatesToJsonRepr,
+#[derive(Deserialize)]
+struct EventWithRelatesToDeHelper<C> {
+    #[serde(rename = "m.relates_to")]
+    relates_to: Option<RelatesToDeHelper>,
```

Comment:
Can we make this a boxed `RawJsonValue` and try different deserializers on that?

Reply:
If this is too much work though we can also merge now and I'll try to revisit myself.

Reply:
I have been trying to do that in different ways, and for some reason whenever I try to deserialize something into a `Box<RawJsonValue>`, it always triggers this error on types that are not `RoomMessageEventContent` (so the extensible events types):

```
Error("invalid type: newtype struct, expected any valid JSON value", line: 1, column: 296)
```